bindings/rust/build.rs

```diff
@@ -1,9 +1,122 @@
+        println!("cargo:rustc-cfg=feature=\"std\"");
+        if target_arch.eq("wasm32") {
+            println!("cargo:rustc-cfg=feature=\"no-threads\"");
+        }
+    }
+    println!("cargo:rerun-if-env-changed=BLST_TEST_NO_STD");
+
+    println!("Using blst source directory {}", blst_base_dir.display());
+
+    // Set CC environment variable to choose alternative C compiler.
+    // Optimization level depends on whether or not --release is passed
+    // or implied.
+
+    #[cfg(target_env = "msvc")]
+    if env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap().eq("32") && !env::var("CC").is_ok() {
+        match std::process::Command::new("clang-cl")
+            .arg("--version")
+            .output()
+        {
+            Ok(out) => {
+                if String::from_utf8(out.stdout)
+                    .unwrap_or("unintelligible".to_string())
+                    .contains("Target: i686-")
+                {
+                    env::set_var("CC", "clang-cl");
+                }
+            }
+            Err(_) => { /* no clang-cl in sight, just ignore the error */ }
+        }
+    }
+
+    let mut cc = cc::Build::new();
+
+    let c_src_dir = blst_base_dir.join("src");
+    println!("cargo:rerun-if-changed={}", c_src_dir.display());
+    let mut file_vec = vec![c_src_dir.join("server.c")];
+
+    if target_arch.eq("x86_64") || target_arch.eq("aarch64") {
+        let asm_dir = blst_base_dir.join("build");
+        println!("cargo:rerun-if-changed={}", asm_dir.display());
+        blst_assembly(&mut file_vec, &asm_dir, &target_arch);
+    } else {
+        cc.define("__BLST_NO_ASM__", None);
+    }
+    match (cfg!(feature = "portable"), cfg!(feature = "force-adx")) {
+        (true, false) => {
+            println!("Compiling in portable mode without ISA extensions");
+            cc.define("__BLST_PORTABLE__", None);
+        }
+        (false, true) => {
+            if target_arch.eq("x86_64") {
+                println!("Enabling ADX support via `force-adx` feature");
+                cc.define("__ADX__", None);
+            } else {
+                println!("`force-adx` is ignored for non-x86_64 targets");
+            }
+        }
+        (false, false) =>
+        {
+            #[cfg(target_arch = "x86_64")]
+            if target_arch.eq("x86_64") && std::is_x86_feature_detected!("adx") {
+                println!("Enabling ADX because it was detected on the host");
+                cc.define("__ADX__", None);
+            }
+        }
+        (true, true) => panic!("Cannot compile with both `portable` and `force-adx` features"),
+    }
+    if env::var("CARGO_CFG_TARGET_ENV").unwrap().eq("msvc") {
+        cc.flag("-Zl");
+    }
+    cc.flag_if_supported("-mno-avx") // avoid costly transitions
+        .flag_if_supported("-fno-builtin")
+        .flag_if_supported("-Wno-unused-function")
+        .flag_if_supported("-Wno-unused-command-line-argument");
+    if target_arch.eq("wasm32") {
```

Comment:
Would the [default CFLAGS for blst](https://github.com/supranational/blst/blob/e9dfc5ee724b5b25d50a3b6226bee8c2c9d5e65d/build.sh#L26) work here?

```
CFLAGS=${CFLAGS:--O2 -fno-builtin -fPIC -Wall -Wextra -Werror}
```

Particularly, `-fPIC -Wall -Wextra -Werror`. Though `-fPIC` shouldn't be enabled on Windows systems.

Reply:
most of these are enabled by `cc` by default, which is what the current version (having `blst` as a dependency in the `Cargo.toml` ) does

Reply:
Ah, you're right, I didn't know Rust enabled those by default.

It looks like these aren't on by default though:

* [`warnings_into_errors`](https://docs.rs/cc/latest/cc/struct.Build.html#method.warnings_into_errors)
* `-fPIC`

Reply:
Eh disregard this. If blst isn't doing it either, it should be fine.